File: iteration2/src/Message.java
import java.time.LocalDateTime;

public class Message {
    private String senderID; // Person who sent the message
    private String receiverID; // Person who received the message

    private String description; // Message content
    private String subject; // Message subject

    private LocalDateTime timeSent; // Time the message was sent

    // Constructor
    public Message(String senderID, String receiverID, String description, String subject) {
        this.senderID = senderID;
        this.receiverID = receiverID;
        this.description = description;
        this.subject = subject;
        this.timeSent = LocalDateTime.now();
    }

    // Getters and Setters
    public String getSenderID() {
        return senderID;
    }

    public String getReceiverID() {
        return receiverID;
    }

    public String getDescription() {
        return description;
    }

    public String getSubject() {
        return subject;
    }

    public LocalDateTime getTimeSent() {
        return timeSent;
    }

    public void setSenderID(String senderID) {
        this.senderID = senderID;
    }

    public void setReceiverID(String receiverID) {
        this.receiverID = receiverID;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

}

File: iteration3/src/message.py
from datetime import datetime


class Message:
    def __init__(self, sender_id, receiver_id, description, subject):
        self.__sender_id = sender_id
        self.__receiver_id = receiver_id
        self.__description = description
        self.__subject = subject
        self.__time_sent = datetime.now()

    def get_sender_id(self):
        return self.__sender_id

    def get_receiver_id(self):
        return self.__receiver_id

    def get_description(self):
        return self.__description

    def get_subject(self):
        return self.__subject

    def get_time_sent(self):
        return self.__time_sent

    def set_sender_id(self, sender_id):
        self.__sender_id = sender_id

    def set_receiver_id(self, receiver_id):
        self.__receiver_id = receiver_id

    def set_description(self, description):
        self.__description = description

    def set_subject(self, subject):
        self.__subject = subject

    def set_time_sent(self, time_sent):
        self.__time_sent = time_sent
